build-tools/automation/yaml-templates/stage-linux-tests.yaml

```diff
@@ -0,0 +1,87 @@
+        downloadPath: $(System.DefaultWorkingDirectory)/bin/Test$(XA.Build.Configuration)
+
+    - template: /build-tools/automation/yaml-templates/run-nunit-tests.yaml@self
+      parameters:
+        testRunTitle: Xamarin.Android.Build.Tests - Linux BuildTest
+        testAssembly: $(System.DefaultWorkingDirectory)/bin/Test$(XA.Build.Configuration)/$(DotNetStableTargetFramework)/Xamarin.Android.Build.Tests.dll
+        dotNetTestExtraArgs: --filter "Name = BuildTest"
+
+    - template: /build-tools/automation/yaml-templates/upload-results.yaml@self
+      parameters:
+        configuration: $(XA.Build.Configuration)
+        artifactName: Test Results - MSBuild - Linux 1
+
+    - template: /build-tools/automation/yaml-templates/fail-on-issue.yaml@self
+
+  - job: linux_tests_smoke_2
+    displayName: Linux > Tests > MSBuild 2
+    pool:
+      name: MAUI-1ESPT
+      image: $(LinuxPoolImage1ESPT)
+      os: linux
+    timeoutInMinutes: 180
+    workspace:
+      clean: all
+    steps:
+    - template: /build-tools/automation/yaml-templates/setup-test-environment.yaml@self
+
+    - task: DownloadPipelineArtifact@2
+      inputs:
+        artifactName: $(TestAssembliesArtifactName)
+        downloadPath: $(System.DefaultWorkingDirectory)/bin/Test$(XA.Build.Configuration)
+
+    - template: /build-tools/automation/yaml-templates/run-nunit-tests.yaml@self
+      parameters:
+        testRunTitle: Xamarin.Android.Build.Tests - Linux PackagingTest
+        testAssembly: $(System.DefaultWorkingDirectory)/bin/Test$(XA.Build.Configuration)/$(DotNetStableTargetFramework)/Xamarin.Android.Build.Tests.dll
+        dotNetTestExtraArgs: --filter "Name = PackagingTest"
+
+    - template: /build-tools/automation/yaml-templates/run-nunit-tests.yaml@self
+      parameters:
+        testRunTitle: Xamarin.Android.Build.Tests - Linux XASdkTests
+        testAssembly: $(System.DefaultWorkingDirectory)/bin/Test$(XA.Build.Configuration)/$(DotNetStableTargetFramework)/Xamarin.Android.Build.Tests.dll
+        dotNetTestExtraArgs: --filter "Name = XASdkTests & Name != XamarinLegacySdk"
+
+    - template: /build-tools/automation/yaml-templates/run-nunit-tests.yaml@self
+      parameters:
+        testRunTitle: Xamarin.Android.Build.Tests - Linux AndroidDependenciesTests
+        testAssembly: $(System.DefaultWorkingDirectory)/bin/Test$(XA.Build.Configuration)/$(DotNetStableTargetFramework)/Xamarin.Android.Build.Tests.dll
+        dotNetTestExtraArgs: --filter "Name = AndroidDependenciesTests"
+
+    - task: ShellScript@2
+      displayName: Test dotnet-local.sh
+      inputs:
+        scriptPath: dotnet-local.sh
+        args: build samples/HelloWorld/HelloWorld/HelloWorld.DotNet.csproj
+
+    - template: /build-tools/automation/yaml-templates/upload-results.yaml@self
+      parameters:
+        configuration: $(XA.Build.Configuration)
+        artifactName: Test Results - MSBuild - Linux 2
+
+    - template: /build-tools/automation/yaml-templates/fail-on-issue.yaml@self
```

Comment:
On second glance, when moving these steps into a new template we'll need to remove the `@ self` suffix.

Reply:
👍 It apparently works fine with them there, but it looks nicer without them.  😄

Reply:
Yeah this pipeline file will work fine with that annotation, but we'd run into issues with the `@ self` reference if we tried to use this new `stage-linux-tets.yaml` template from a different pipeline file. That `@ self` suffix is a requirement from 1ES pipeline templates that I don't fully understand, but it's only needed in the top level pipeline file and preferably shouldn't be used elsewhere from what I've seen so far.